docs/GetStarted/cmake_options_and_variables.md

```diff
@@ -0,0 +1,79 @@
+# IREE-specific CMake Options and Variables
+
+This gives a brief explanation of IREE specific CMake options and variables.
```

Comment:
We could expand on this overview section, taking inspiration from https://llvm.org/docs/CMake.html
(that doesn't have to happen in this PR)

Reply:
Yes, I was looking into that one before starting (also I don't like the layout :wink:), but I wasn't sure how far we should document the options. So far we only document the IREE specific ones, somehow equivalent to https://llvm.org/docs/CMake.html#llvm-specific-variables. In addition we have one MLIR specific variable which is `MLIR_DIR`. For that one introducing a new section seemed overloaded. But indeed, we can create sections like https://llvm.org/docs/CMake.html#frequently-used-cmake-variables and could further add explanations or links to the appropriate docs for options like `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER`.